freestyle-cache/shared/src/main/scala/hashmap/Hasher.scala

```diff
@@ -0,0 +1,21 @@
+package freestyle.cache.hashmap
+
+trait Hasher[A] {
+  def hashCode(a: A): Int
+}
+
+object Hasher {
```

Comment:
We can include here automatic instances for any product type such as case classes, `HList`, `TupleN` etc with shapeless generic derivation given there is also an implicit `Hasher` for the contained elements.